core/resume_hook.cc

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2014-2016, The Regents of the University of California.
+// Copyright (c) 2016-2017, Nefeli Networks, Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation
+// and/or other materials provided with the distribution.
+//
+// * Neither the names of the copyright holders nor the names of their
+// contributors may be used to endorse or promote products derived from this
+// software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include "resume_hook.h"
+
+#include <map>
+#include <string>
+#include <utility>
+
+namespace bess {
+
+// Worker resume hooks -------------------------------------------------------
+template <>
+std::map<std::string, ResumeHookFactory<Worker>>
+    &ResumeHookFactory<Worker>::all_resume_hook_factories_holder(bool reset) {
+  // Maps from hook names to hook factories. Tracks all hooks (via their
+  // ResumeHookFactorys).
+  static std::map<std::string, ResumeHookFactory<Worker>>
+      all_resume_hook_factories;
+
+  if (reset) {
```

Comment:
I can't figure out where reset is ever true (but this is the same as the gate hook code, where I also can't figure out where this is ever true, so at least I'm consistently missing something 😺 )

Reply:
`reset` is here to facilitate testing. `ModuleBuilder::all_module_builders_holder` has a similar flag that I could have sworn used to be used in some tests, but I can't find them anymore either